validator/validator.c

```diff
@@ -1848,7 +1860,9 @@ processValidate(struct module_qstate* qstate, struct val_qstate* vq,
 			"of trust to keys for", vq->key_entry->name,
 			LDNS_RR_TYPE_DNSKEY, vq->key_entry->key_class);
 		vq->chase_reply->security = sec_status_bogus;
-		errinf(qstate, "while building chain of trust");
+		vq->chase_reply->reason_bogus = LDNS_EDE_DNSSEC_BOGUS;
+		errinf_ede(qstate, "while building chain of trust",
+			LDNS_EDE_DNSKEY_MISSING);
```

Comment:
Why the discrepancy between setting `vq->chase_reply->reason_bogus` and `errinf_ede()`?

```suggestion
		vq->chase_reply->reason_bogus = LDNS_EDE_DNSKEY_MISSING;
		errinf_ede(qstate, "while building chain of trust",
			LDNS_EDE_DNSKEY_MISSING);
```